config/generateRunFile.coffee

```diff
@@ -424,6 +424,124 @@ generateDev = (KONFIG, options) ->
+        export CHANGE_MINIKUBE_NONE_USER=true
+        sudo -E minikube start --vm-driver=none # for local development: minikube start --disk-size 50g --cpus 2 --memory 5120
+
+        sleep 25
+
+        sudo sed --in-place --expression 's@path: \/opt\/koding@path: '"$(pwd)"'@g' deployment/kubernetes/workers-pod/containers.yaml
+        sudo sed --in-place --expression 's@$(hostname)@'"$(hostname)"'@g' deployment/kubernetes/workers-pod/containers.yaml
+        kubectl create -f $KONFIG_PROJECTROOT/deployment/kubernetes/namespace.yaml
+        kubectl create -f $KONFIG_PROJECTROOT/deployment/kubernetes/workers-pod/containers.yaml
+
+        sleep 5
+        echo "checking if pod is ready"
+        check_pod_ready
+        export WORKERS_POD_NAME="$(kubectl get pods --namespace service-dev -o jsonpath="{.items[0].metadata.name}")"
+        sleep 10
+
+        k8s_health_check $WORKERS_POD_NAME service-dev 5 120 backend-services
+
+        # guest user can only connect via localhost, a test user equivalent to guest will be created to check connectivity via POD IP.
+        # more info on guest user update: https://www.rabbitmq.com/blog/2014/04/02/breaking-things-with-rabbitmq-3-3/
+        k8s_health_check $WORKERS_POD_NAME service-dev 5 120 rabbitmq
+        sleep 25
+        kubectl exec -n service-dev $WORKERS_POD_NAME -c rabbitmq -- bash -c "rabbitmqctl add_user test test && rabbitmqctl set_user_tags test administrator && rabbitmqctl set_permissions -p / test '.*' '.*' '.*'"
+
+        k8s_connectivity_check $WORKERS_POD_NAME service-dev backend-services
+
+        echo "all services are ready"
+
+      else
+        # TODO: stop the cluster, maybe even have an option to kill the cluster?
+        minikube stop
+      fi
+    }
+
+    function k8s_connectivity_check () {
+      # args: $1: POD NAME, $2: NAMESPACE, $3: container name
+      kubectl exec -n $2 $1 -c $3 -- bash -c "./run is_ready"
+    }
+
+    function k8s_health_check () {
+      # args: $1: POD NAME, $2: NAMESPACE, $3: health check interval (recommended 5 seconds), $4: timeout duration
+      declare interval=$3
+      declare timeout=$4
+      declare duration=0
+
+      sleep $interval
+
+      declare response_code=$(kubectl exec -n $2 $1 -c $5 -- echo 'i am alive')
+
+      echo -n $5 : 'health-check : '
+
+      until [[ $response_code != *"error"* ]]; do
+        if [ $duration -eq $timeout ]; then
+          echo ' timed out!'
+          exit 255
+        fi
+
+        echo -n '.'
+
+        sleep $interval
+        duration=$((duration + interval))
+
+        declare response_code=$(kubectl exec -n $2 $1 -c $5 -- echo 'i am alive')
+      done
+
+      echo ' succeeded!'
+    }
+
+    function check_pod_ready () {
+      declare interval=5
+      declare timeout=600
+      declare duration=0
+      declare response_code=$(kubectl get pods --namespace service-dev -o jsonpath="{.items[0].status.phase}")
+
+      until [[ $response_code != *"Pending"* ]]; do
```

Comment:
How about invoking a function which basically returns 0 or 1 depending on the response code?